adapter/src/utils/useLocale.js

```diff
@@ -23,22 +52,41 @@ const setGlobalLocale = (locale) => {
     }
     moment.locale(locale)
 
-    const simplifiedLocale = simplifyLocale(locale)
-    i18n.changeLanguage(simplifiedLocale)
+    const resolvedLocale = validateLocaleByBundle(locale)
+    i18n.changeLanguage(resolvedLocale)
+
+    console.log('🗺 Global d2-i18n locale initialized:', resolvedLocale)
+}
+
+// Sets the global direction based on the app's configured direction
+// (which should be done to affect modals, alerts, and other portal elements),
+// then returns the locale's direction for use on the header bar
+const handleDirection = ({ locale, configDirection }) => {
```

Comment:
this method is doing two distinct things: getting the global direction and setting the root direction, and the name is confusing its responsibility .. maybe the root direction can be set inline in the hook, and this is just a pure function that returns the global direction.